src/main/scala/Engine.scala

```diff
@@ -3,11 +3,13 @@ package org.template.classification
 import io.prediction.controller.EngineFactory
 import io.prediction.controller.Engine
 
-class Query(
-  val attr0 : Double,
-  val attr1 : Double,
-  val attr2 : Double
-) extends Serializable
+class Query extends Serializable{
+  var attrs: Seq[Double] = null
+  def this(attrs: Double*) {
+    this()
+    this.attrs = attrs
+  }
+}
```

Comment:
Hey @mzjgithub , wondering why we need to have an auxiliary constructor here. Can we simply have

```scala
class Query(val attrs: Seq[Double]) extends Serializable
```

?

Reply:
Up to you